### PR TITLE
Implementation of bionic tiny stack

### PR DESCRIPTION
**Pull request by fg-j**
## Summary

Adds initial stack definition and tests for the Bionic Tiny Stack. This stack should be functionally identical to what's published in the paketo-buildpacks/tiny-stack-release repo.

## Differences from Legacy Bionic Tiny Stack Implementation
Stack image filesystem diffs generated with [container-diff](https://github.com/GoogleContainerTools/container-diff)
### Build
```
container-diff diff paketobuildpacks/build:1.3.55-tiny-cnb daemon://updated-bionic-build:latest --type file | tee pbcopy

-----File-----

These entries have been added to paketobuildpacks/build:1.3.55-tiny-cnb: None

These entries have been deleted from paketobuildpacks/build:1.3.55-tiny-cnb:
FILE                          SIZE
/etc/group-                   446B
/etc/gshadow-                 374B
/etc/passwd-                  926B
/etc/shadow-                  501B
/etc/subgid-                  0
/etc/subuid-                  0
/home/cnb/.bash_logout        220B
/home/cnb/.bashrc             3.7K
/home/cnb/.profile            807B

These entries have been changed between paketobuildpacks/build:1.3.55-tiny-cnb and updated-bionic-build:latest:
FILE                                 SIZE1         SIZE2
/var/log/lastlog                     285.4K        28.8K
/var/log/dpkg.log                    246.6K        246.6K
/var/log/tallylog                    62.6K         6.3K
/var/log/faillog                     31.3K         3.2K
/var/log/apt/term.log                25.9K         25.9K
/var/log/apt/history.log             12.6K         12.6K
/var/log/apt/eipp.log.xz             9.2K          9.2K
/var/cache/ldconfig/aux-cache        8.2K          8.2K
/var/log/alternatives.log            6.3K          6.3K
/var/lib/apt/extended_states         4K            4K
/etc/shadow                          526B          501B
/etc/gshadow                         382B          374B
/etc/apt/sources.list                225B          208B
/etc/default/locale                  92B           92B
/etc/subuid                          17B           0
/etc/subgid                          17B           0
```

- There are a couple of differences that stem from `jam create-stack`  **not** using `useradd` to create the `cnb` user:
  -  the user does not have a default`.bashrc`, `.profile`, or `.bash_logout` files in its home directory. This means that aliases like `ls -al ==> ll` are not available by default in the new build images. (See [StackOverflow](https://askubuntu.com/questions/971836/at-what-point-is-the-bashrc-file-created) for discussion of this `useradd` behaviour.)
  - `/etc/subuid` and `/etc/subgid` are empty because [`useradd`](https://man7.org/linux/man-pages/man8/useradd.8.html) populates these automatically
 
- `/etc/apt/sources.list` is different because we no longer use the `multiverse` package repository component. There were no packages on the tiny image that required that source.
- `/etc/shadow` and `/etc/gshadow` are encrypted versions of `/etc/passwd` and `/etc/group`, respectively. Since the unencrypted files have no diff, I'm not worried about the encrypted versions being slightly different.
- `/var/lib/apt/extended_states` is modified by `apt-get` and [`apt-mark`](https://manpages.ubuntu.com/manpages/bionic/man8/apt-mark.8.html). The diff reflects the fact that `libexpat1` is **NOT** explicitly installed in the updated version of the bionic tiny stack build image. It's auto-installed as a transitive dependency of `git`.
- The files suffixed with a dash (e.g. `/etc/group-`) are simply [backups of these important `/etc` files](https://manpages.ubuntu.com/manpages/kinetic/en/man5/shadow.5.html). It's ok that they're not included in the new stack.
- the `/etc/default/locale` file routinely changes between stack builds
- log and cache files routinely change between builds of stacks; nothing to see here

### Run
```
container-diff diff paketobuildpacks/run:1.3.55-tiny-cnb daemon://updated-bionic-run:latest --type file

-----File-----

These entries have been added to paketobuildpacks/run:1.3.55-tiny-cnb: None

These entries have been deleted from paketobuildpacks/run:1.3.55-tiny-cnb:
FILE                        SIZE
/var/lib/dpkg/status        1B

These entries have been changed between paketobuildpacks/run:1.3.55-tiny-cnb and updated-bionic-run:latest:
FILE                                          SIZE1        SIZE2
/var/lib/dpkg/status.d/openssl                1.1K         1006B
/var/lib/dpkg/status.d/libc6                  1K           963B
/var/lib/dpkg/status.d/ca-certificates        990B         883B
/var/lib/dpkg/status.d/libssl1.1              972B         875B
/var/lib/dpkg/status.d/base-files             911B         813B
/var/lib/dpkg/status.d/tzdata                 858B         767B
/var/lib/dpkg/status.d/zlib1g                 715B         616B
/etc/os-release                               452B         509B
/var/lib/dpkg/status.d/netbase                445B         372B
/etc/passwd                                   193B         190B
/etc/group
```
- `/var/lib/dpkg/status` was empty in the old bionic run image. It's been intentionally removed.
- The `status.d/*` files are different because we write slightly different metadata into the files for each dependency. A notable difference is that we no longer include upstream source info (`Source-Package`, `Source-Version`, and `Source-Upstream-Version` fields)
- the `/etc/os-release` file is different since the "support URL" and other human-readable information about the stack has changed. See my other comment on the PR.
- some changes caused by not using `useradd`:
  - the change in the `/etc/passwd` field is the removal of the User ID Info (GECOS) field in the file. This is an optional, "comment" field.
  - the change in `/etc/group` is the removal of a trailing colon, with no effect on behaviour
  
## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+				MatchRegexp("Version: [a-z0-9\\.\\-\\:]+ubuntu[0-9\\.]+"),
+				ContainSubstring("Architecture: amd64"),
+			)))
+
+			Expect(image).NotTo(HaveFile("/usr/share/ca-certificates"))
+
+			Expect(image).To(HaveFileWithContent("/etc/os-release", SatisfyAll(
+				// TODO: This is different from what we currently ship. Are we ok
+				// with changing?
+				ContainSubstring(`PRETTY_NAME="Paketo Buildpacks Tiny Bionic"`),
```
Comment (fg-j) [inline metadata_test.go:256]: Question for @paketo-buildpacks/stacks-maintainers  – I've updated the contents of the `os-release` file to be a bit more up-to-date compared to what we [ship in the current Tiny stack](https://github.com/paketo-buildpacks/stacks/blob/53374cadbf79a64f2290708337aaa0174f24fc14/tiny/dockerfile/run/files/os-release).

Comment (sophiewigmore) [reply metadata_test.go:256]: I think this is OK. I'm extremely surprised to see the Cloud Foundry references in the current Tiny stack and I'd like to get away from that. I can't imagine that is breaking information either.

**Comment by sophiewigmore** [comment]
@fg-j  Re: the differences in the stack images.

From your research, are any of these differences red flags to you in that they may change behavior for users? Or is more research needed to understand that? From your (**very helpful**) descriptions, the `status.d/*` differences in the run image are of interest to understand ramification, and additionally I’d also maybe want to check that the lack of a `.bashrc`, `.profile`, and/or .`bash_logout` files have no effect.

**Comment by sophiewigmore** [comment]
RE your TODO list:` create custom push-image.yml workflow that pushes to the appropriate repos/tags; add push-image.yml to .syncignore`

We'll need the same for base and full Bionic stacks too. I wonder if it would make more sense to extend the existing github-config stack `push-buildpackage` workflow to perform this special case for any `bionic-*` stacks?

**Comment by fg-j** [comment]
> From your research, are any of these differences red flags to you in that they may change behavior for users? 
All in all, I don't see any red flags in the diff.

The default of `.profile`, `.bashrc` and `.bash_logout` all impact the behaviour of the `bash` shell in the build image, but *_only when the shell is running in interactive mode_*. This means a user might see a difference if they `exec` into a build/run container using `bash`, but it won't impact how bash scripts (including bash buildpacks) run in the build container. I don't think it's a big deal, since our stacks/app containers aren't intended to provide an interactive `bash` experience.

The change in the `status.d` files makes `syft` generate slightly different package metadata when it scans the run image. All the package versions and key information is the same, but syft notices the source version metadata when it's present. I don't think this is a huge deal.

File: .github/.syncignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CODEOWNERS
+workflows/push-legacy-image.yml
```
Comment (sophiewigmore) [inline .github/.syncignore:2]: let's use the newly add modified push-image workflow instead as discussed

File: buildpack_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/paketo-buildpacks/occam"
+	. "github.com/paketo-buildpacks/occam/matchers"
+	"github.com/paketo-buildpacks/packit/v2/pexec"
+)
+
+const lifecycleVersion = "0.14.0"
```
Comment (sophiewigmore) [inline buildpack_integration_test.go:20]: I see you already have a TODO for this that's checked off, but let's stop using hardcoded lifecycle

File: buildpack_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  build-image = "%s:latest"
+  id = "io.paketo.stacks.tiny"
+  run-image = "%s:latest"
+`,
+			stack.BuildImageID,
+			stack.RunImageID,
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(docker.Pull.Execute(fmt.Sprintf("buildpacksio/lifecycle:%s", lifecycleVersion))).To(Succeed())
```
Comment (sophiewigmore) [inline buildpack_integration_test.go:79]: this can go too

File: stack/stack.toml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    libssl1.1 \
+    netbase \
+    openssl \
+    tzdata \
+    zlib1g \
+    """
+
+[deprecated]
+  legacy-sbom = true
+  mixins = true
```
Comment (sophiewigmore) [inline stack/stack.toml:57]: do we want these fields? I don't think we ship with a legacy SBOM now

Comment (sophiewigmore) [reply stack/stack.toml:57]: Yup, you're right actually!

**Pull request merged**